engine/src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala

```diff
@@ -105,28 +111,45 @@ class WorkflowManagerActor(config: Config,
   private val tag = self.path.name
 
   private var abortingWorkflowToReplyTo = Map.empty[WorkflowId, ActorRef]
+  private var shutdownHookThreadOption: Option[ShutdownHookThread] = None
 
   override def preStart(): Unit = {
     addShutdownHook()
     // Starts the workflow polling cycle
     self ! RetrieveNewWorkflows
   }
 
+  override def postStop() = {
+    // If the actor is stopping, then there's nothing to wait for later. Remove the shutdown hook.
+    tryRemoveShutdownHook()
```

Comment:
ToL:

Hmm, in this case either we've already stopped everything (so the shutdown hook would be a no-op) or there are jobs left running (and we'd want to run the shutdown hook, assuming it was added)

So, I think we'd prefer to not have this line?